libraries/chain/include/eosio/chain/authority.hpp

```diff
@@ -221,19 +220,22 @@ struct authority {
 
 
 struct shared_authority {
-   explicit shared_authority( chainbase::allocator<char> alloc )
-   :keys(alloc),accounts(alloc),waits(alloc){}
-
-   shared_authority& operator=(const authority& a) {
-      threshold = a.threshold;
-      keys.clear();
-      keys.reserve(a.keys.size());
-      for(const key_weight& k : a.keys) {
-         keys.emplace_back(shared_key_weight::convert(keys.get_allocator(), k));
-      }
-      accounts = decltype(accounts)(a.accounts.begin(), a.accounts.end(), accounts.get_allocator());
-      waits = decltype(waits)(a.waits.begin(), a.waits.end(), waits.get_allocator());
-      return *this;
+   shared_authority() = default;
+
+   shared_authority(const authority& auth) :
+      threshold(auth.threshold),
+      keys(auth.keys),
+      accounts(auth.accounts),
+      waits(auth.waits)
+   {
+   }
+
+   shared_authority(authority&& auth) :
```

Comment:
`explicit`

Reply:
This makes some assignments a little more verbose, but OK.

Reply:
Leaving as-is because the type of the constructor parameter leave little chance for unexpected conversions.